src/battery-check/battery-check.c

```diff
@@ -0,0 +1,149 @@
+        };
+
+        static const struct option options[] = {
+                { "help",    no_argument, NULL, 'h'         },
+                { "version", no_argument, NULL, ARG_VERSION },
+                {}
+        };
+
+        int c;
+
+        assert(argc >= 0);
+        assert(argv);
+
+        while ((c = getopt_long(argc, argv, "h", options, NULL)) >= 0)
+
+                switch (c) {
+
+                case 'h':
+                        help();
+                        return 0;
+
+                case ARG_VERSION:
+                        return version();
+
+                case '?':
+                        return -EINVAL;
+
+                default:
+                        assert_not_reached();
+                }
+
+        if (optind < argc)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                       "%s takes no argument.",
+                                       program_invocation_short_name);
+        return 1;
+}
+
+static int run(int argc, char *argv[]) {
+        int r;
+
+        log_parse_environment();
+        log_open();
+
+        r = parse_argv(argc, argv);
+        if (r <= 0)
+                return r;
+
+        r = battery_is_discharging_and_low();
+        if (r < 0) {
+                log_warning_errno(r, "Failed to check battery status, ignoring: %m");
+                return 0;
+        }
+        if (r > 0) {
+                _cleanup_close_ int fd = -EBADF;
+                _cleanup_free_ char *message = NULL;
+
+                if (asprintf(&message, "%s Battery level critically low. Please connect your charger or the system will power off in 10 seconds.", special_glyph(SPECIAL_GLYPH_LOW_BATTERY)) < 0)
+                        return log_oom();
+                log_emergency("%s", message);
+                fd = open_terminal("/dev/console", O_WRONLY|O_NOCTTY|O_CLOEXEC);
+                if (fd < 0)
+                        log_warning_errno(fd, "Failed to open console, ignoring: %m");
+                else
+                        dprintf(fd,  ANSI_HIGHLIGHT_RED "%s" ANSI_NORMAL "\n", message);
+                battery_check_send_plymouth_message(message);
+                sleep(10);
+                r = battery_is_discharging_and_low();
+                if (r > 0) {
+                        log_emergency("%s", message);
+                        return r;
+                }
+                if (r < 0) {
+                        if (asprintf(&message, "%s Battery level critically low. Current battery level could not be read, ensure to plug in and retry. Powering off.", special_glyph(SPECIAL_GLYPH_LOW_BATTERY)) < 0)
+                                return log_oom();
+                        log_emergency("%s", message);
+                        dprintf(fd,  ANSI_HIGHLIGHT_RED "%s" ANSI_NORMAL "\n", message);
+                        battery_check_send_plymouth_message(message);
+                }
+                        sleep(10);
```

Comment:
hmm, why another sleep(10?

Reply:
I was looking at allowing another 10 seconds for the user to see the second message, but as you pointed out sir, we should just take the second reading to be identical to the first.
I'll drop it definitely.

Perhaps I was thinking of the function calls to be as unpredictable as Nigerian telco network, where a line can connect successfully this second and fail to connect the next few seconds 😁😁.